### DISCUSSION TITLE
Best way to backup Oxigraph server data

### DISCUSSION THREAD

**Discussion by PiotrNowara** [oxigraph/oxigraph#556] [Q&A] [answered]
Hi, 

I'm evaluating Oxigraph for archiving RDF data dumps from another triple store. My Oxigraph is running on k8s with a mounted persistent storage. What's the best way to backup Oxigraph server data just in case the storage gets corrupted? Is there some admin endpoint or a command to trigger an automatic bacckup or I will have to write some custom code to extract the data and copy it to files? 

Thanks,
Piotr

**Comment by Tpt** [comment, marked as answer]
Yes, `./oxigraph_server backup --location directory_to_backup_from --destination directory_to_backup_to` to backup using Oxigraph internal storage format and `./oxigraph_server dump --location directory_to_backup_from --file file_to_backup_to.nq` to backup in N-Quads and move data to an other RDF system. Both can work while the server is running.

**Comment by PiotrNowara** [comment]
Hi @Tpt ,

thanks, this looks good. I just couldn't find it when typing: oxigraph_server --help which only shows: serve, load and help.

One thing more: how I can run this command when running Oxigraph in Docker? The issue is I'm not able to access bash when running it in Docker.

`PS C:\Users\piotr> docker exec -it 3914574546bc bash

OCI runtime exec failed: exec failed: unable to start container process: exec: "bash": executable file not found in $PATH: unknown`

I'm starting it like this:
`docker run --it -v $PWD/data:/data -p 7878:7878 ghcr.io/oxigraph/oxigraph --location /data serve --bind 0.0.0.0:7878`

Thanks,
Piotr

**Reply by Tpt** [reply]
> thanks, this looks good. I just couldn't find it when typing: oxigraph_server --help which only shows: serve, load and help.

This command have been introduced in v0.3.14. Are you using an older version?

> The issue is I'm not able to access bash when running it in Docker.

Yes, the Docker image is as small as possible so does not ship usual unix utilities.

I believe you be able to directly call `oxigraph_server`. For example:
```bash
docker run -d -v $PWD/data:/data -v $PWD/backup:/backup -p 7878:7878 ghcr.io/oxigraph/oxigraph --location /data serve --bind 0.0.0.0:7878
docker exec -it 05a199ab10cc oxigraph_server backup --location /data --destination /backup/test
```

**Comment by PiotrNowara** [comment]
My bad, I had an old version installed via cargo and I checked the old one instead of new one.

I tried out the script you sent me and it works without any issues. 

Now I wonder about logs:

1. could you tell me where Oxigraph stores its logs?
2. is there any way to configure logging level sensitivity, for example to check if the backup is done?

Thanks,
Piotr

**Reply by Tpt** [reply]
Currently Oxigraph does not do any logging outside of what RocksDB is doing (it's something I need to improve). RocksDB logs are in the `LOG` files in the storage directory. So, there is no way to see if/when backup is done from the log. You might just want to check the return number from the `backup` command.
